xmtp_mls/migrations/2024-12-18-170645_add_dm_id/up.sql

```diff
@@ -0,0 +1,26 @@
+ALTER TABLE groups ADD COLUMN dm_id TEXT;
+ALTER TABLE groups ADD COLUMN last_message_ns BIGINT NOT NULL DEFAULT ((strftime('%s', 'now') * 1000000000) + (strftime('%f', 'now') * 1000000));
+
+-- Fill the dm_id column
+UPDATE groups
+SET dm_id = 'dm:' ||
+    LOWER(
+        CASE
+            WHEN LOWER((SELECT inbox_id FROM identity)) < LOWER(dm_inbox_id)
+            THEN (SELECT inbox_id FROM identity) || ':' || dm_inbox_id
+            ELSE dm_inbox_id || ':' || (SELECT inbox_id FROM identity)
+        END
+    )
```

Comment:
Going to be writing a test to ensure this does what's expected before merging.

Reply:
Do we still need this database migration? https://github.com/xmtp/libxmtp/pull/1374 Rich suggested we could just us the dm_inbox_id?

Reply:
Having a consistent key across DM's that should be stitched makes querying much simpler.